ThirdParty/json.swift

```diff
@@ -58,33 +58,33 @@ extension JSON {
     }
     /// constructs JSON object from string
     public convenience init(string:String) {
-        let enc:NSStringEncoding = NSUTF8StringEncoding
-        self.init(data: string.dataUsingEncoding(enc)!)
+        let enc:String.Encoding = String.Encoding.utf8
+        self.init(data: string.data(using: enc)!)
     }
     /// parses string to the JSON object
     /// same as JSON(string:String)
-    public class func parse(string:String)->JSON {
+    public class func parse(_ string:String)->JSON {
         return JSON(string:string)
     }
     /// constructs JSON object from the content of NSURL
-    public convenience init(nsurl:NSURL) {
-        var enc:NSStringEncoding = NSUTF8StringEncoding
+    public convenience init(nsurl:URL) {
+        var enc:String.Encoding = String.Encoding.utf8
         do {
             let str = try String(NSString(
-                contentsOfURL:nsurl, usedEncoding:&enc))
+                contentsOf:nsurl, usedEncoding:&enc.rawValue))
             self.init(string:str)
         } catch let error as NSError {
             self.init(error)
         }
     }
     /// fetch the JSON string from NSURL and parse it
     /// same as JSON(nsurl:NSURL)
-    public class func fromNSURL(nsurl:NSURL) -> JSON {
+    public class func fromNSURL(_ nsurl:URL) -> JSON {
         return JSON(nsurl:nsurl)
     }
     /// constructs JSON object from the content of URL
     public convenience init(url:String) {
-        if let nsurl = NSURL(string:url) as NSURL? {
+        if let nsurl = URL(string:url) as URL? {
```

Comment:
no need to cast to `URL?` here. The result of `init(string:)` is already optional